internal/queue/consumer.go

```diff
@@ -0,0 +1,117 @@
+// Package queue implements the message queue functionality using RabbitMQ.
+package queue
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Konstantsiy/image-converter/internal/config"
+	"github.com/Konstantsiy/image-converter/internal/converter"
+	"github.com/Konstantsiy/image-converter/internal/repository"
+	"github.com/Konstantsiy/image-converter/internal/storage"
+	"github.com/Konstantsiy/image-converter/pkg/logger"
+	"github.com/streadway/amqp"
+)
+
+// Consumer listens to the queue and processes outgoing messages.
+type Consumer struct {
+	client  *rabbitMQClient
+	repo    *repository.Repository
+	storage *storage.Storage
+}
+
+func NewConsumer(repo *repository.Repository, storage *storage.Storage, conf *config.Config) (*Consumer, error) {
+	client, err := initRabbitMQClient(conf)
+	if err != nil {
+		return nil, err
+	}
+	return &Consumer{repo: repo, storage: storage, client: client}, nil
+}
+
+// ListenToQueue listens to the queue channel in a separate goroutine.
+func (c *Consumer) ListenToQueue() error {
+	err := c.client.ch.Qos(1, 0, false)
+	if err != nil {
+		return fmt.Errorf("can't configure QoS: %w", err)
+	}
+
+	msgChannel, err := c.client.ch.Consume(c.client.queue.Name, "", false, false, false, false, nil)
+	if err != nil {
+		return fmt.Errorf("can't register channel: %w", err)
+	}
+
+	go func() {
+		for {
+			msg := <-msgChannel
+			go c.consumeFromQueue(&msg)
+		}
+	}()
+
+	return nil
+}
+
+// consumeFromQueue wraps the message processing and confirms its completion.
+func (c *Consumer) consumeFromQueue(msg *amqp.Delivery) {
+	var data queueMessage
+	err := json.NewDecoder(bytes.NewReader(msg.Body)).Decode(&data)
+	if err != nil {
+		logger.Error(context.Background(), fmt.Errorf("can't decode queue message: %w", err))
+		return
+	}
```

Comment:
the message doesn't get acknowledged, is that an error?
I'd suggest for you to return an `error` from `consumeFromQueue` and move positive and negative acknowledgments to the caller. And probably logging as well